tests/experimental/pydantic/schema/test_defaults.py

```diff
@@ -156,3 +157,34 @@ def b(self, user: StrawberryUser) -> str:
     """
 
     assert print_schema(schema) == textwrap.dedent(expected).strip()
+
+
+@needs_pydanticv2
+def test_v2_explicit_default():
+    class User(pydantic.BaseModel):
+        name: Optional[str]
+
+    @strawberry.experimental.pydantic.type(User, all_fields=True)
+    class PydanticUser:
+        ...
+
+    @strawberry.type
+    class Query:
+        @strawberry.field
+        def a(self) -> PydanticUser:
+            ...
```

Comment:
I think if you do this codecov won't complain

```suggestion
            raise NotImplementedError
```